### ISSUE TITLE
Cannot read property 'version' of undefined when trying to access an attached storage account

### ISSUE THREAD

**Issue by jgangel** [microsoft/AzureStorageExplorer#1961]
**Storage Explorer Version**: 1.10.1
**Build Number**: 20190919.2
**Platform/OS**: Windows 10
**Architecture**: x64
**Regression From**:  Version from 2 weeks ago, not sure which was it.

## Bug Description ##
When trying to access an attached storage account or container, the message Cannot read property 'version' of undefined appears.

## Steps to Reproduce ##

Open Storage Explorer
Open Local & Attached/Storage Accounts
Open any of the attached storage account

## Expected Experience ##
Storage account containers appearing and navigate through them

## Actual Experience ##
The following message appears as pop-up:
Unable to load attached Storage Account. Error: Cannot read property 'version' of undefined

**Comment by MRayermannMSFT** [comment]
@jgangel , are you seeing a broken node or is this message appearing in a dialog when you expand "Storage Accounts"? Can you maybe share some screenshots of what you're seeing, step by step?

**Comment by jgangel** [comment]
It happens when I try to open a broken node. The actual problem is that all nodes broke up for apparently no reason, 1 or 2 weeks ago. I'll try the solution in the troubleshooting guide to check if that helps.

Here is a screenshot of my Storage Explorer

![image](https://user-images.githubusercontent.com/42182153/65904678-8cdd1c80-e384-11e9-841c-6f1eccb2e5c9.png)

**Comment by lloyd-dotmodus** [comment]
Copied over from closed ticket:

> Hi @JasonYeMSFT, I've tried de and re-attaching. I've re-added and the credentials don 't seem to be coming up in the Keychain credentials now. I did also look at the developer tools but there weren't any errors there after adding a connection.

I see @craxal has marked this ticket as duplicate. I'll comment on the #1961 ticket then.

Regards

**Comment by lloyd-dotmodus** [comment]
So I think my issue on the above is the fact that the keychain credentials are not stored automatically, and the dialog box for authentication according to the troubleshooting guide never comes up.

> 6. You're prompted with a message like "Service hub wants to access the Keychain." Enter your Mac admin account password and select Always Allow (or Allow if Always Allow isn't available).

https://docs.microsoft.com/en-us/azure/storage/common/storage-explorer-troubleshooting?tabs=Windows%2C1904#mac-keychain-errors

Can I get some help on how to restore the keychain credentials for storage explorer on a mac?

Regards

**Comment by lloyd-dotmodus** [comment]
I have deleted the Storage Explorer folder in /Library/Application Support as per suggestions on the troubleshooting guides and this ticket: https://github.com/Microsoft/AzureStorageExplorer/issues/552 in order to try and retrigger the authentication window, but still nothing.

**Comment by craxal** [comment]
@lloyd-dotmodus Did you ever click "Always Allow" when the dialog popped up in the past? You may need to go into Keychain yourself and remove any Storage Explorer-related passwords. Keychain should also let you remove whitelisted applications (see [here](https://apple.stackexchange.com/questions/116211/how-can-i-undo-always-allow-in-keychain-item)).

**Comment by lloyd-dotmodus** [comment]
I think I may have but I've removed all Storage Explorer-related passwords from the keychain several times. I just need to figure out how to re-add them manually again, cause that prompt doesn't come up and when I open the storage explorer they don't get added automatically. I'll try re-install again the latest version and see if the keychain picks it up, hopefully a prompt will appear.

**Comment by craxal** [comment]
@lloyd-dotmodus I don't know if manually entering anything will be much help. Storage Explorer generates an encryption key and stores it in your password manager. If your connection info isn't encrypted with this key, things obviously won't work as expected.

**Comment by lloyd-dotmodus** [comment]
Okay so somehow or other when I rebooted today and re-opened up the storage explorer again that I installed previously, the keychain application passwords got added all of a sudden (Same process did nothing earlier in the week). I didn't actually change or add anything manually on the keychain access app, and I had rebooted and restarted the application before to no avail, but now somehow they just there. So I can't say what the fix is for anyone else that may have this same issue. I'm just glad it works now. So I guess my issue is resolved now. Thanks for the feedback and help, much appreciated.  Hopefully the thread still helps someone else in need.

Regards
